app/src/main/scala/org/alephium/explorer/service/HashrateService.scala

```diff
@@ -93,19 +93,22 @@ case object HashrateService extends StrictLogging {
       } yield ()
     )
 
-  private def findLatestHashrate(intervalType: IntervalType): DBActionR[Option[HashrateEntity]] =
-    HashrateSchema.table
-      .filter(_.intervalType === intervalType)
-      .sortBy(_.timestamp.desc)
-      .result
-      .headOption
+  private def findLatestHashrate(intervalType: IntervalType)(implicit
+      ec: ExecutionContext
+  ): DBActionR[Option[TimeStamp]] =
+    sql"""
+      SELECT block_timestamp
+      FROM hashrates
+      WHERE interval_type = $intervalType
+      ORDER BY block_timestamp DESC
+    """.asAS[TimeStamp].headOrNone
```

Comment:
Question: Does this query fetch multiple rows from the DB and return only the first row (`headOption`) after reading multiple rows? Would adding `LIMIT 1` to the query be helpful here?

Reply:
good point, indeed that's better so we don't grab all data from DB just to keep only the first value.

I'll make a quick check in all our codebase, we might have similar places